contracts/UniversalRouter.sol

```diff
@@ -42,7 +42,9 @@ contract UniversalRouter is RouterImmutables, IUniversalRouter, Dispatcher, Rewa
             (success, output) = dispatch(command, input);
 
             if (!success && successRequired(command)) {
-                revert ExecutionFailed({commandIndex: commandIndex, message: output});
+                assembly {
+                    revert(add(32, output), mload(output))
```

Comment:
> which wouldn't register as a custom error

yeah i havent come up with a way for it to become a custom error sadly (without it being an argument in another custom error).

My motivation was basically that `'ExecutionFailed(0, "0x7dbe7e89")'` is so horrible to try to read (both for cases where the hex string is a custom error, or when it isnt), because it doesnt parse the hex.

Whereas if you _just_ output the hex - then it either gets parsed as a readable custom error, _or_ as a readable string. And then users can actually see what the message is.

But I see your point that we have some long strings. I'm gonna go through and make those strings shorter 👍

Reply:
Yes will also add a `0` length check. 

> also we're eating the commandIndex of the error now

Yeah I thought about that too - I just concluded that seeing `InvalidSignature` without knowing the command index is more helpful than seeing `ExecutionFailed(0, "0x8baa579f")` where you know its command 0 but have no idea what actually went wrong

Reply:
I was wrong, I forgot I bubble custom errors - so am turning that cryptopunks error into a custom one

Reply:
do you know if etherscan et al, will understand the custom error if it's not a part of the contract abi?

Reply:
honestly it looks like etherscan doesnt show custom errors at all yet...
if you look at [this tx](https://etherscan.io/tx/0x1200a35ecb0d47cdb4ad008c9c857857f5ac56fc23c00c04056bbce02ca91f80) etherscan just says it reverted
but if you look at it on [tenderly](https://dashboard.tenderly.co/tx/mainnet/0x1200a35ecb0d47cdb4ad008c9c857857f5ac56fc23c00c04056bbce02ca91f80) you can see it was a `revert TooLittleReceived()` in the V2 router

Reply:
so even our existing custom errors dont work for users of etherscan 🤪

Reply:
> think either way devs can figure out what went wrong since there's decoding tools

To be clear no platforms support custom errors yet... so i dont think people can access the parameters of custom errors (what we currently have) nor bubbled custom errors (what I've done).

> Would this break for any [integrations] that do not [use custom errors]?

No it would bubble their string error, in the same way it bubbles custom errors

> fwiw etherscan does show plaintext errors which is what we have here currently.

 What we have currently is not a plaintext error - its a custom error so no users can see it on etherscan 😭 

Now that I know neither of these options will be viewable by our users on etherscan/tenderly i'm happy to revert my change lol - it doesnt really make any difference 😆

Reply:
Note: etherscan does support custom errors: https://twitter.com/etherscan/status/1452955443991494661
I've also used to see custom errors thrown by the contracts I write. They clearly don't show up for Universal Router though, idk why